### ISSUE TITLE
FAILED: akka.http.scaladsl.server.HttpAppSpec.HttpApp should start without ActorSystem

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#824]
https://jenkins.akka.io:8498/job/pr-validator-akka-http/704/testReport/junit/akka.http.scaladsl.server/HttpAppSpec/HttpApp_should_start_without_ActorSystem/

```
Error Message

The code passed to eventually never returned normally. Attempted 150 times over 5.030236196999999 seconds. Last failure message: The future returned an exception of type: akka.stream.StreamTcpException, with message: Tcp command [Connect(localhost:35085,None,List(),Some(10 seconds),true)] failed..

Stacktrace

org.scalatest.exceptions.TestFailedDueToTimeoutException: The code passed to eventually never returned normally. Attempted 150 times over 5.030236196999999 seconds. Last failure message: The future returned an exception of type: akka.stream.StreamTcpException, with message: Tcp command [Connect(localhost:35085,None,List(),Some(10 seconds),true)] failed..
	at org.scalatest.concurrent.Eventually$class.tryTryAgain$1(Eventually.scala:421)
	at org.scalatest.concurrent.Eventually$class.eventually(Eventually.scala:439)
	at akka.http.scaladsl.server.HttpAppSpec.eventually(HttpAppSpec.scala:26)
	at akka.http.scaladsl.server.HttpAppSpec.akka$http$scaladsl$server$HttpAppSpec$$callAndVerify(HttpAppSpec.scala:202)
	at akka.http.scaladsl.server.HttpAppSpec$$anonfun$1$$anonfun$apply$mcV$sp$2.apply(HttpAppSpec.scala:64)
	at akka.http.scaladsl.server.HttpAppSpec$$anonfun$1$$anonfun$apply$mcV$sp$2.apply(HttpAppSpec.scala:52)
	at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:1078)
	at org.scalatest.TestSuite$class.withFixture(TestSuite.scala:196)
	at akka.testkit.AkkaSpec.withFixture(AkkaSpec.scala:57)
	at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:1075)
	at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
	at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
	at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:1088)
	at akka.testkit.AkkaSpec.runTest(AkkaSpec.scala:57)
	at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
	at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:396)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
	at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:373)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:410)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
	at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:379)
	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
	at org.scalatest.WordSpecLike$class.runTests(WordSpecLike.scala:1147)
	at akka.testkit.AkkaSpec.runTests(AkkaSpec.scala:57)
	at org.scalatest.Suite$class.run(Suite.scala:1147)
	at akka.testkit.AkkaSpec.org$scalatest$WordSpecLike$$super$run(AkkaSpec.scala:57)
	at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
	at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
	at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
	at org.scalatest.WordSpecLike$class.run(WordSpecLike.scala:1192)
	at akka.testkit.AkkaSpec.org$scalatest$BeforeAndAfterAll$$super$run(AkkaSpec.scala:57)
	at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:213)
	at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:210)
	at akka.testkit.AkkaSpec.run(AkkaSpec.scala:57)
	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:472)
	at sbt.TestRunner.runTest$1(TestFramework.scala:76)
	at sbt.TestRunner.run(TestFramework.scala:85)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
	at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:185)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
	at sbt.TestFunction.apply(TestFramework.scala:207)
	at sbt.Tests$$anonfun$9.apply(Tests.scala:216)
	at sbt.Tests$$anonfun$9.apply(Tests.scala:216)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.scalatest.exceptions.TestFailedException: The future returned an exception of type: akka.stream.StreamTcpException, with message: Tcp command [Connect(localhost:35085,None,List(),Some(10 seconds),true)] failed.
	at org.scalatest.concurrent.Futures$FutureConcept$class.tryTryAgain$1(Futures.scala:523)
	at org.scalatest.concurrent.Futures$FutureConcept$class.futureValueImpl(Futures.scala:550)
	at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValueImpl(ScalaFutures.scala:275)
	at org.scalatest.concurrent.Futures$FutureConcept$class.futureValue(Futures.scala:476)
	at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValue(ScalaFutures.scala:275)
	at akka.http.scaladsl.server.HttpAppSpec$$anonfun$akka$http$scaladsl$server$HttpAppSpec$$callAndVerify$1.apply(HttpAppSpec.scala:204)
	at akka.http.scaladsl.server.HttpAppSpec$$anonfun$akka$http$scaladsl$server$HttpAppSpec$$callAndVerify$1.apply(HttpAppSpec.scala:202)
	at org.scalatest.concurrent.Eventually$class.makeAValiantAttempt$1(Eventually.scala:395)
	at org.scalatest.concurrent.Eventually$class.tryTryAgain$1(Eventually.scala:409)
	... 72 more
Caused by: akka.stream.StreamTcpException: Tcp command [Connect(localhost:35085,None,List(),Some(10 seconds),true)] failed
```

**Comment by jlprat** [comment]
Facts and discoveries so far:
* Exact same test runs for the Java API and works.
* Fails on rare occasions (1 so far)
* The server with port 35085 looks like is shutdown 6 milliseconds after being started
* Shutting down the server should only happen after querying the server
* The logs in Jenkins look incomplete as there are only 4 occurrences of `Server online at` String (written when server is started). There should be 6 of them (3 for the Scala API tests, 3 for the Java ones)

Is it possible that a port collision has happened?

**Comment by jrudolph** [comment]
Thanks for doing the research here, @jlprat. If a port collision is likely it might make sense to wait for changes of #803 which would allow to bind on port 0 to make the OS choose the server port.

**Comment by jlprat** [comment]
I can wait for it, then modify the tests to use port `0` and let's see if it happens again

**Comment by jrudolph** [comment]
Cool, thanks.

**Comment by jrudolph** [comment]
Fixed in #849.

**Comment by jonas** [comment]
#824 merged so closing this
